src/application.ts

```diff
@@ -21,6 +23,11 @@ export interface ApplicationOptions {
   resolver: Resolver;
 }
 
+export interface InstanceInitializer {
+  name?: string;
+  initialize(app: RegistryWriter): void;
+}
```

Comment:
I'm going to propose that we call this interface simply `Initializer`, even though it currently happens once per application instance. I don't think we will be carrying forward Ember's `ApplicationInstance` class here, so the name `InstanceInitializer` seems to fit less well. Plus, by accepting only a `RegistryWriter` and not allowing container access, the `initialize` signature aligns with Ember's plain `Initializer`.